approved/v93k/flow_control.tf

```diff
@@ -1725,38 +1795,24 @@ test_flow
   run_and_branch(pgm_BEA7F3B)
   then
   {
-    @T5_BEA7F3B_RAN = 1;
+    run(read0_BEA7F3B);
   }
   else
   {
     stop_bin "", "fail", , bad, noreprobe, red, 10, over_on;
     @T5_BEA7F3B_RAN = 1;
   }
-  if @T5_BEA7F3B_RAN == 1 then
-  {
-    run(read0_BEA7F3B);
-  }
-  else
-  {
-  }
   print_dl("Test the block form of if_ran");
   run_and_branch(pgm_1_BEA7F3B)
   then
   {
-    @T6_BEA7F3B_RAN = 1;
-  }
-  else
-  {
-    stop_bin "", "fail", , bad, noreprobe, red, 10, over_on;
-    @T6_BEA7F3B_RAN = 1;
-  }
-  if @T6_BEA7F3B_RAN == 1 then
-  {
     run(read0_1_BEA7F3B);
     run(read0_2_BEA7F3B);
   }
   else
   {
+    stop_bin "", "fail", , bad, noreprobe, red, 10, over_on;
+    @T6_BEA7F3B_RAN = 1;
   }
```

Comment:
This doesn't seem quite right.
The intention of if_ran is that the parent test should be set to continue on fail and a _RAN flag should be set whether it passed or failed.
The dependent test should therefore run whether the parent test passed or failed, but here it now runs only if the parent test passes and it sets a now redundant flag in the fail case.
The original implementation doesn't look quite right either as the stop_bin should have been removed.

You could just remove the RAN flags from your optimization, checking for auto_generated and name containing _RAN should be good enough.